tutorials/examples/import-demo-1.dart

```diff
@@ -0,0 +1,204 @@
+// Demonstrates import of just "pointycastle.dart"
+
+import 'dart:convert';
+import 'dart:math';
+import 'dart:typed_data';
+
+import "package:pointycastle/pointycastle.dart";
+
+void main() {
+  usingRegistry();
+  // explicit();
+}
+
+void usingRegistry() {
+  final sha256 = Digest("SHA-256");
+  final sha1 = Digest("SHA-1");
+  final md5 = Digest("MD5");
+
+  final _digest = sha256.process(_data);
+
+  final hmacSha256 = Mac("SHA-256/HMAC");
+  final hmacSha512 = Mac("SHA-512/HMAC");
+  final hmacMd5 = Mac("MD5/HMAC");
+
+  final _hmacValue = hmacSha256.process(_data);
+
+  final kd = KeyDerivator("SHA-256/HMAC/PBKDF2");
+
+  final _sGen = Random.secure();
+  final _seed =
+  Uint8List.fromList(List.generate(32, (n) => _sGen.nextInt(255)));
+  final secRnd = SecureRandom("Fortuna")..seed(KeyParameter(_seed));
+
+  // AES-CBC encryption
+
+  final _salt = secRnd.nextBytes(32);
+
+  final keyDerivator256 = KeyDerivator("SHA-256/HMAC/PBKDF2")
+    ..init(Pbkdf2Parameters(_salt, 10000, 256 ~/ 8));
+
+  final aes256key = keyDerivator256.process(_secret);
+
+  final _iv = secRnd.nextBytes(128 ~/ 8);
+  final aesCbc = BlockCipher("AES/CBC")
+    ..init(true, ParametersWithIV(KeyParameter(aes256key), _iv));
+
+  final _paddedData = Uint8List(
+      _data.length + (aesCbc.blockSize - (_data.length % aesCbc.blockSize)))
+    ..setAll(0, _data);
+  Padding("PKCS7").addPadding(_paddedData, _data.length);
+
+  final _ciphertext = aesCbc.process(_paddedData);
+
+  // RSA key generation and signing
+
+  final keyGen = KeyGenerator("RSA");
+  keyGen.init(ParametersWithRandom(
+      RSAKeyGeneratorParameters(BigInt.parse('65537'), 2048, 64), secRnd));
+  final _pair = keyGen.generateKeyPair();
+
+  final signer = Signer("SHA-256/RSA")
+    ..init(true, PrivateKeyParameter<RSAPrivateKey>(_pair.privateKey));
+
+  final _signature = signer.generateSignature(_data) as RSASignature;
+
+  final verifier = Signer("SHA-256/RSA")
+    ..init(false, PublicKeyParameter<RSAPublicKey>(_pair.publicKey));
+  final sigOk = verifier.verifySignature(_data, _signature);
+
+  print('''
+Data: "${utf8.decode(_data)}"
+
+SHA-256: ${bin2hex(_digest)}
+SHA-1:   ${bin2hex(sha1.process(_data))}
+MD5:     ${bin2hex(md5.process(_data), separator: ':')}
+
+HMAC-SHA256: ${bin2hex(_hmacValue)}
+HMAC-512:    ${bin2hex(hmacSha512.process(_data))}
+HMAC-MD5:    ${bin2hex(hmacMd5.process(_data))}
+
+AES-CBC ciphertext:
+${bin2hex(_ciphertext, wrap: 64)}
+
+Signature:
+${bin2hex(_signature.bytes, wrap: 64)}
+Verifies: $sigOk
+''');
+}
+
+/*
```

Comment:
Why is this commented out?

Reply:
That is the function with the code that creates the classes using their _constructors_ rather than via the registry. It is commented out because demo 1 only imports "pointycastle.dart", so that code won't even compile/run because the implementation classes have not been imported.

The code has not been deleted, so that you can run "diff" with the other import demos and quickly see the changed import statements, rather than being distracted by lots of missing lines of code.

I was trying to figure out what the difference is between importing "pointycastle.dart", vs "export.dart", vs "api.dart" plus individual libraries. I couldn't find any behaviour differences between importing just "export.dart" and "api.dart" plus individual libraries. The original README claimed in the later the registry would only work for the implementations that have been explicitly imported. But I found the registry still worked for implementations that have not been explicitly imported. I also couldn't find any size differences: compiling to native produced the same executable size, and compiling to JavaScript produced almost the same sizes too. Maybe programs start up faster?

The import-demo-?.dart examples are still very rough. I gave up trying to find the answer, so that I can make the pull request. Delete the import-demo-?.dart examples if you want.